nova/compute/manager.py

```diff
@@ -1730,8 +1730,11 @@ def _get_device_name_for_instance(self, instance, bdms, block_device_obj):
             return self.driver.get_device_name_for_instance(
                 instance, bdms, block_device_obj)
         except NotImplementedError:
-            return compute_utils.get_device_name_for_instance(
-                instance, bdms, block_device_obj.get("device_name"))
+            @utils.synchronized(instance.uuid + "-bdms")
```

Comment:
This lock is only used here and only used to serialize multiple threads trying to get a device-name for a BDM for an instance at the same time, so we don't give out a device name twice?

Reply:
Exactly. It might be better handled by a db constraint, but my understanding is, that those are more difficult to maintain out of tree.